src/main/java/com/solvd/micro9/users/messaging/KfProducer.java

```diff
@@ -1,23 +1,12 @@
 package com.solvd.micro9.users.messaging;
 
+import com.solvd.micro9.users.domain.aggregate.User;
 import com.solvd.micro9.users.domain.es.Es;
-import lombok.RequiredArgsConstructor;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.kafka.core.reactive.ReactiveKafkaProducerTemplate;
-import org.springframework.stereotype.Component;
 
-@Component
-@RequiredArgsConstructor
-public class KfProducer {
+public interface KfProducer {
```

Comment:
@eyakauleva I mean something like this:
```java
interface Producer<K, X> {
void send(K key, X val);
```

```java
class UserProducer implements Producer<String, User> {
@Override
void send(final String key, final User user) {
}
}
```

WDYT?

Reply:
@h1alexbel fixed acfcad3